runelite-client/src/main/java/net/runelite/client/plugins/driftnet/DriftNetPlugin.java

```diff
@@ -0,0 +1,223 @@
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package net.runelite.client.plugins.driftnet;
+
+import com.google.inject.Provides;
+import lombok.Getter;
+import net.runelite.api.*;
+import net.runelite.api.events.*;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.eventbus.Subscribe;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.ui.overlay.OverlayManager;
+import javax.inject.Inject;
+import java.util.HashMap;
+import java.util.HashSet;
+
+@PluginDescriptor(
+	name = "DriftNet Helper",
+	description = "Provides various overlays to assist with underwater driftnet fishing.",
+	tags = {"overlay", "fishing"}
+)
+
+public class DriftNetPlugin extends Plugin
+{
+	private final int NORTH_NET_ID = 31433;
+	private final int SOUTH_NET_ID = 31434;
+
+	DriftNet northNet = new DriftNet();
+	DriftNet southNet = new DriftNet();
+
+	private NPC interacting = null;
+
+	@Getter
+	private HashMap<NPC, Long> fishes = new HashMap<>();
+
+	@Inject
+	private OverlayManager overlayManager;
+
+	@Inject
+	private DriftNetConfig config;
+
+	@Inject
+	private DriftNetOverlay driftNetOverlay;
+
+	@Inject
+	private InfoBoxOverlay infoBoxOverlay;
+
+
+	@Inject
+	private Client client;
+
+	@Getter
+	private HashSet<NPC> bosses = new HashSet<>();
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		overlayManager.add(driftNetOverlay);
+		overlayManager.add(infoBoxOverlay);
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		overlayManager.remove(driftNetOverlay);
+		overlayManager.remove(infoBoxOverlay);
+	}
+
+	@Provides
+	DriftNetConfig provideConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(DriftNetConfig.class);
+	}
+
+	@Subscribe
+	public void onGameObjectSpawned(GameObjectSpawned event)
+	{
+		GameObject gameObject = event.getGameObject();
+		//ObjectComposition objectComposition = client.getObjectDefinition(gameObject.getId());
+		if (event.getGameObject().getId() == SOUTH_NET_ID)
+		{
+			southNet.setDriftNet(gameObject);
+		}
+		else if (event.getGameObject().getId() == NORTH_NET_ID)
+		{
+			northNet.setDriftNet(gameObject);
+		}
+	}
+
+	@Subscribe
+	public void onMenuEntryAdded(MenuEntryAdded event)
```

Comment:
The Drift nets state is controlled by 4 varbits, 2 for each. The first (5812 for north, 5814 for south) controls the visual state of the net, and the other two (5813 and 5815) control the fish caught. These should be used instead of trying to so stuff with the menu entries, since menu stuff only changes when you hover over it

Reply:
Hi thanks a lot for pointing this out. For future reference, how does one figure out what varbits are responsible for controlling which variables?  How were you able to figure out these varbits?

Reply:
If you enable the [devtools](https://github.com/runelite/runelite/wiki/Using-the-client-developer-tools), you can use the Var Inspector to see what changes